File: python/algo/202601/algo3767.py
from typing import List

# 一、贪心策略 - 1.1 从最小/最大开始贪心
class Solution:
    def maxPoints(self, technique1: List[int], technique2: List[int], k: int) -> int:
        # 自行解答
        ans = 0
        for i, [t1, t2] in enumerate(sorted(zip(technique1, technique2), key=lambda p:p[1]-p[0])):
            if i < k: ans += t1
            else: ans += max(t1, t2)
        return ans

File: cplus/leetcode/202601/algo3767.cpp
#include <bits/stdc++.h>
using namespace std;

// 一、贪心策略 - 1.1 从最小/最大开始贪心
class Solution {
public:
    long long maxPoints(vector<int>& technique1, vector<int>& technique2, int k) {
        // 自行解答 按照 technique1[i] - technique2[i] 倒序排列
        int n = technique1.size(), idx[n];
        iota(idx, idx+n, 0);
        sort(idx, idx + n, [&](int i, int j) {
            return technique1[i]-technique2[i] > technique1[j]-technique2[j];
        });
        long long ans = 0;
        for (int i=0; i<n; ++i) {
            auto id = idx[i];
            if (i < k) ans += technique1[id];
            else ans += max(technique1[id], technique2[id]);
        }

        return ans;
    }
};